### PR TITLE
Fixes #5787 - Make ManagedSelector report better JMX data.

### LINKED ISSUE #5787: Make ManagedSelector report better JMX data
**Jetty version**
9.4.x

**Description**
`ManagedSelector` should be enhanced to report useful information via JMX.

For example:
1. Number of select() wakeups
2. How many keys were selected for each wakeup
3. How many keys each selector manages
4. How long it takes to process selected keys
5. How long select() operations lasts (i.e. how long it's "idle")
6. Etc.

### PR DESCRIPTION
**Pull request by sbordet**
Added SampleStatistics to record data and JMX methods to export it.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

File: jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java [hunk 1]
```diff
@@ -144,6 +148,36 @@ protected void doStop() throws Exception
         super.doStop();
     }
 
+    @ManagedAttribute(value = "Total number of keys", readonly = true)
+    public int getTotalKeys()
```
Comment (lorban) [inline jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java:152]: Rename this `getTotalSelectableKeys` maybe?

File: jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java [hunk 2]
```diff
@@ -144,6 +148,36 @@ protected void doStop() throws Exception
...
+    }
+
+    @ManagedAttribute(value = "Total number of select() calls", readonly = true)
+    public long getSelectCount()
+    {
+        return _keyStats.getCount();
+    }
+
+    @ManagedOperation(value = "Resets the statistics", impact = "ACTION")
+    public void resetStats()
```
Comment (lorban) [inline jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java:176]: Are you _really sure_ we want to expose this?

Comment (gregw) [reply jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java:176]: I think it is good to have and easy to ignore if you don't want to use it.

**Review by gregw — APPROVED** [review]
LGTM

**Pull request merged**